decoder/semantic_tokens.go

```diff
@@ -135,6 +165,34 @@ func (d *PathDecoder) tokensForExpression(expr hclsyntax.Expression, constraints
 			}
 		}
 
+		address, _ := lang.TraversalToAddress(eType.AsTraversal())
+		countIndexAttr := lang.Address{
+			lang.RootStep{
+				Name: "count",
+			},
+			lang.AttrStep{
+				Name: "index",
+			},
+		}
+		countAvailable := icontext.ActiveCountFromContext(ctx)
+		// TODO why is countAvailable not true here?
+		log.Printf("Found Expression: %q / %q - %v+", countAvailable, address.Equals(countIndexAttr), address)
+		// if address.Equals(countIndexAttr) && countAvailable {
+		if address.Equals(countIndexAttr) {
+			traversal := eType.AsTraversal()
+			tokens = append(tokens, lang.SemanticToken{
+				Type:      lang.TokenTraversalStep,
+				Modifiers: []lang.SemanticTokenModifier{},
+				Range:     traversal[0].SourceRange(),
+			})
+			tokens = append(tokens, lang.SemanticToken{
+				Type:      lang.TokenTraversalStep,
+				Modifiers: []lang.SemanticTokenModifier{},
+				Range:     traversal[1].SourceRange(),
```

Comment:
One minor problem with this is that it's misaligned with how we report address segments in all other contexts, that is that we leave out the initial dot.
<img width="613" alt="Screenshot 2022-10-10 at 10 35 20" src="https://user-images.githubusercontent.com/287584/194884048-b9299ffa-b7af-4ece-be9f-417bbea7fd81.png">
![Screenshot 2022-10-10 at 15 02 36](https://user-images.githubusercontent.com/287584/194884140-cb694146-7f6f-41bb-b114-ce81a97df10a.png)

See LOC 230-235 for how we account for this.

We kinda expect the separator to not have a special meaning in the context of semantic highlighting.

To put it another way, the dot is what makes the address address, whether it is a "valid" address worth highlighting (because it points to a block or attribute which actually exists) depends on the individual segments, although the dot is important pre-requisite. A more practical reason is that users will likely want the dots to be highlighted differently from the segments.

Reply:
Addressed!